### PR TITLE
Update Conversation service

### PR DESCRIPTION
**Pull request by glennrfisher**
(Note: Builds off the changes from #678.)

This pull request updates the Conversation service to use generated code and the `Codable` protocol.

To do before merging:
- [x] Check code coverage
- [ ] Resolve test error with logs
- [x] Update `DialogNode` model to reflect optional properties
- [x] Generate with latest Conversation swagger

### CODE REVIEW

File: Source/ConversationV1/Models/MessageRequest.swift [hunk 1]
```diff
@@ -17,111 +17,78 @@
...
-     */
-    public init(
-        input: Input? = nil,
-        alternateIntents: Bool? = nil,
-        context: Context? = nil,
-        entities: [Entity]? = nil,
-        intents: [Intent]? = nil,
-        output: Output? = nil)
-    {
+     - parameter alternateIntents: Whether to return more than one intent. Set to `true` to return all matching intents.
```
Comment (mkistler) [inline Source/ConversationV1/Models/MessageRequest.swift]: I guess the only change here is the formatting.  We need to look into reflowing comments (and maybe also code reformatting), but that's outside the scope of this PR.

File: Source/ConversationV1/Models/OutputData.swift [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    private enum CodingKeys: String, CodingKey {
+        case logMessages = "log_messages"
+        case text = "text"
+        case nodesVisited = "nodes_visited"
+        static let allValues = [logMessages, text, nodesVisited]
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let dynamic = try decoder.container(keyedBy: DynamicKeys.self)
```
Comment (mkistler) [inline Source/ConversationV1/Models/OutputData.swift]: The text highlighting here seems to imply that `dynamic` is a Swift keyword.  Not sure we need to do anything about this, but I thought I should mention it.

File: Source/ConversationV1/Models/Counterexample.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import Foundation
+
+/** Counterexample. */
+public struct Counterexample {
+
+    /// The text of the counterexample.
+    public var text: String
+
+    /// The timestamp for creation of the counterexample.
+    public var created: String
```
[the remark this answers is not in the document]
Comment (glennrfisher) [reply Source/ConversationV1/Models/Counterexample.swift]: Good idea. I opened an [issue](https://github.ibm.com/arf/planning-sdk-squad/issues/312).

File: Source/ConversationV1/Models/MessageRequest.swift (hunk 1 above)
Comment (glennrfisher) [reply Source/ConversationV1/Models/MessageRequest.swift]: True. Opened an [issue](https://github.ibm.com/arf/planning-sdk-squad/issues/313).

File: Source/ConversationV1/Models/OutputData.swift (hunk 2 above)
Comment (glennrfisher) [reply Source/ConversationV1/Models/OutputData.swift]: Yup! It is a Swift keyword, although it's only ever used as a modifier in front of a declaration (e.g. `dynamic var` or `dynamic function`). Xcode is smart enough not to highlight this instance.

It's easy enough to change, though. I updated the template, regenerated the models, and pushed a commit.

**Comment by glennrfisher** [comment]
I agree that we should start thinking about a v1.0.0 release. Since some folks have been asking for a better way to get/set context variables, though, I think it would be helpful to push this out sooner.

**Comment by glennrfisher** [comment]
Thanks again for the review, @mkistler! 😃

**Comment by glennrfisher** [comment]
The bug with `listLogs` seems to be a service error. I have opened an issue and am following up. I don't anticipate any changes to our code, so I'm going to go ahead and merge this pull request so that we can publish a new release.

**Pull request merged**